collection/set.go

```diff
@@ -47,23 +49,23 @@ func Intersection(a Set, b Set) Set {
 	return result
 }
 
-type mapSet struct {
+type mapSet[T any] struct {
 	mutex sync.RWMutex
 	// empty interface with nil stored is smaller than bool
 	m map[interface{}]interface{}
 }
```

Comment:
I thought about it but decided not to because there were already many changes and having an interface makes the set more versatile, it can hold different types.

Anyway, I wasn't sure which way to go, I'll update it.